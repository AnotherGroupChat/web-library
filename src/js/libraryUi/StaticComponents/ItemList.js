'use strict';

var log = require('libzotero/lib/Log').Logger('zotero-web-library:ItemsList');

var React = require('react');
var ReactDOM = require('react-dom');

var ItemDetails = require('./ItemDetails.js');

var ItemsList = React.createClass({
	getDefaultProps: function() {
		return {};
	},
	getInitialState: function() {
		return {};
	},
	render: function() {
		return (
			<section id="item-list">
				<div id="item-list-container">
					<header className="touch-header">
						<h3>Collection title</h3>
						<div className="toolbar"></div>
					</header>
					<div>Item List</div>
				</div>
				<ItemDetails />
			</section>
		);
	}
});

module.exports = ItemsList;
