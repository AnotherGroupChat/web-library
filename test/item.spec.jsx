import React from 'react';
import renderer from 'react-test-renderer';

import ItemList from '../src/js/component/item/list';
import itemsFix from './fixtures/items';

it('Renders items list with data', () => {
	let tree = renderer.create(
		<ItemList items={ itemsFix } />
	).toJSON();
	expect(tree).toMatchSnapshot();
});

it('Renders items list empty', () => {
	let tree = renderer.create(
		<ItemList items={ [] } />
	).toJSON();
	expect(tree).toMatchSnapshot();
});

it('Renders items list fetching', () => {
	let tree = renderer.create(
		<ItemList items={ itemsFix } isFetching={ true } />
	).toJSON();
	expect(tree).toMatchSnapshot();
});