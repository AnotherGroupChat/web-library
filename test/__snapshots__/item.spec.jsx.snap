// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders items list empty 1`] = `
<div
  className="item-list-wrap"
>
  <table
    className="item-list-head hidden-touch hidden-sm-down"
  >
    <thead>
      <tr>
        <th>
          Title
        </th>
        <th>
          Creator
        </th>
        <th>
          Year
        </th>
        <th
          className="hidden-touch hidden-sm-down"
        >
          Date Modified
        </th>
        <th
          className="hidden-touch hidden-sm-down"
        />
        <th
          className="hidden-touch hidden-sm-down"
        />
      </tr>
    </thead>
  </table>
  <div
    className="item-list-body"
  >
    <ul
      className="item list"
    />
  </div>
</div>
`;

exports[`Renders items list fetching 1`] = `
<div
  className="spinner"
/>
`;

exports[`Renders items list with data 1`] = `
<div
  className="item-list-wrap"
>
  <table
    className="item-list-head hidden-touch hidden-sm-down"
  >
    <thead>
      <tr>
        <th>
          Title
        </th>
        <th>
          Creator
        </th>
        <th>
          Year
        </th>
        <th
          className="hidden-touch hidden-sm-down"
        >
          Date Modified
        </th>
        <th
          className="hidden-touch hidden-sm-down"
        />
        <th
          className="hidden-touch hidden-sm-down"
        />
      </tr>
    </thead>
  </table>
  <div
    className="item-list-body"
  >
    <ul
      className="item list"
    >
      <li
        className="item "
        onClick={[Function]}
      >
        <div
          className="metadata"
        >
          document-1
        </div>
        <div
          className="metadata author"
        />
        <div
          className="metadata year"
        />
        <div
          className="metadata hidden-touch hidden-sm-down"
        >
          2016-02-02
        </div>
        <div
          className="metadata hidden-touch hidden-sm-down"
        />
        <div
          className="metadata hidden-touch hidden-sm-down"
        />
      </li>
      <li
        className="item "
        onClick={[Function]}
      >
        <div
          className="metadata"
        >
          document-2
        </div>
        <div
          className="metadata author"
        />
        <div
          className="metadata year"
        />
        <div
          className="metadata hidden-touch hidden-sm-down"
        >
          2016-02-03
        </div>
        <div
          className="metadata hidden-touch hidden-sm-down"
        />
        <div
          className="metadata hidden-touch hidden-sm-down"
        />
      </li>
    </ul>
  </div>
</div>
`;
